tools/rsyslogd.c

```diff
@@ -1621,10 +1658,8 @@ initAll(int argc, char **argv)
 	hdlr_enable(SIGCHLD, hdlr_sigchld);
 	hdlr_enable(SIGHUP, hdlr_sighup);
 
-	if(rsconfNeedDropPriv(ourConf)) {
-		/* need to write pid file early as we may loose permissions */
-		CHKiRet(writePidFile());
-	}
+	CHKiRet(checkStartupOK());
```

Comment:
in the case of non-priv drop, this now writes the pid file earlier than before. That can cause races in some exotic situations. I know the chance is slim, but I remember cases where we had this in the testbench, for example.

Reply:
Thank you @rgerhards - apologies for delays on my side as well, these are interesting times. I will refresh my memory on this code path and look at what should be done. It would be interesting to understand that race condition better too. Do you happen to recall which testbench test caused the race?

Reply:
Hello @rgerhards - hope you are well. Sorry for the delay, but I have pushed a change that puts the non-priv drop parts of the code path back the way it was. Thank you.